layersvt: DevSim 1.0.5 code reorgs and bug fix

mikew-lunarg opened the pull request:
A collection of small changes to fix a bug, but don't otherwise affect existing functionality, in preparation for new functionality in development.

1) Simplify array checking code, since lengths are enforced by schema validation.

2) GetArray() now returns int for the number of elements in an array as read from the JSON contents.

3) Minor change to test driver script, to simplify debugging.

4) Try again to get the desired results from JsonCpp's type system.  Use the source, not the docs.

5) Fix gold file for 64bit values
    
Change-Id: I2860f21e643b93d259cf364c0d13e51a6c3b1f17

mikew-lunarg commented:
Ping.  The changes are pretty small.

courtney-g commented on tests/devsim_layer_test.sh:
```diff
@@ -23,26 +23,31 @@ export LD_LIBRARY_PATH=${PWD}/../loader:${LD_LIBRARY_PATH}
+FILENAME_01_STDOUT="device_simulation_layer_test_1.txt"
+
+export VK_DEVSIM_FILENAME="${PWD}/${FILENAME_01_IN}"
+${VKJSON_INFO} > ${FILENAME_01_STDOUT}
```
Curious, why include ${PWD} above, but not here?

mikew-lunarg replied:
Fair point, I think that may be historical.  There may have been more cd-ing going on previously, that would have made locating vkjson_info trickier.

courtney-g replied:
Make it consistent then?

mikew-lunarg replied:
done

mikew-lunarg commented:
Made several fixes per feedback:
- adjusted test driver script
- fixed type identification from JSON
- fixed expected 64-bit values in gold file.

cnorthrop commented:
Latest commits clears things up, thanks Mike!
